src/protagonist/Test.Helpers/Integration/DatabaseTestDataPopulation.cs

```diff
@@ -207,4 +210,34 @@ public static class DatabaseTestDataPopulation
         asset.Result.Entity.WithTestThumbnailMetadata(metadataValue);
         return asset;
     }
+
+    public static ValueTask<EntityEntry<Asset>> WithTestDeliveryChannel(
+        this ValueTask<EntityEntry<Asset>> asset,
+        string deliveryChannel,
+        int? policyId = null)
+    {
+        if (!AssetDeliveryChannels.IsValidChannel(deliveryChannel))

```

Comment:
Good call adding helper methods for test setups, can save a lot of boiler plate. One question on this check - is this required? As these methods are purely used to generate test data, could a valid test involve having invalid delivery-channels?